src/many-ledger/src/migration.rs

```diff
@@ -1,76 +1,16 @@
-pub mod data;
-
-use merk::Op;
-use std::{collections::BTreeSet, fmt::Debug};
-use tracing::info;
-
-#[cfg(feature = "migrate_blocks")]
-use many_protocol::ResponseMessage;
-
-use crate::storage::data::MIGRATIONS_KEY;
-
-#[cfg(feature = "block_9400")]
-mod block_9400;
+use linkme::distributed_slice;
+use many_error::ManyError;
+use many_migration::{InnerMigration, Migration};
+use std::collections::BTreeMap;
 
-#[cfg(feature = "migrate_blocks")]
-pub fn migrate(tx_id: &[u8], response: ResponseMessage) -> ResponseMessage {
-    match hex::encode(tx_id).as_str() {
-        #[cfg(feature = "block_9400")]
-        "241e00000001" => block_9400::migrate(response),
-        _ => response,
-    }
-}
-
-#[typetag::serde(tag = "type")]
-pub trait Migration: Debug + Send + Sync {
-    fn migrate(&self, persistent_store: &mut merk::Merk) -> Vec<(Vec<u8>, Op)>;
-    fn block_height(&self) -> u64;
-    fn issue(&self) -> Option<&str>;
-    fn name(&self) -> &str;
-}
-
-impl PartialEq<Box<dyn Migration>> for Box<dyn Migration> {
-    fn eq(&self, other: &Box<dyn Migration>) -> bool {
-        self.name().eq(other.name())
-    }
-}
-
-impl Eq for Box<dyn Migration> {}
+pub mod block_9400;
+pub mod data;
 
-impl PartialOrd<Box<dyn Migration>> for Box<dyn Migration> {
-    fn partial_cmp(&self, other: &Box<dyn Migration>) -> Option<std::cmp::Ordering> {
-        self.name().partial_cmp(other.name())
-    }
-}
+pub type LedgerMigrations<'a> = BTreeMap<&'a str, Migration<'a, merk::Merk, ManyError>>;
```

Comment:
Just make this `'static`. Will save you a lot of references around (and reduce your delta quite a bit). WDYT?